### PR TITLE
Add Verifier developer guide to presto doc 

### PR DESCRIPTION
**Pull request by caithagoras**

### CODE REVIEW

File: presto-docs/src/main/sphinx/develop/verifier-mechanics.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
+==================
+Verifier Mechanics
+==================
+
+The Presto Verifier can be used to test two versions Presto against each other.
```
[the remark this answers is not in the document]
Comment (mayankgarg1990) [reply presto-docs/src/main/sphinx/develop/verifier-mechanics.rst]: Add some context around what kind of tests. Example - `Verifier validates if the outputs of Presto queries remain the same across different versions. In addition to that, by using the query ids, one can perform different forms of analysis, including but not limited to performance analysis`.

File: presto-docs/src/main/sphinx/develop/verifier-mechanics.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Procedures and Functionalities
+------------------------------
+
+The verifier verifies a set of given queries in the following procedures.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/develop/verifier-mechanics.rst]: maybe - `The following steps summarize the working of verifier`

File: presto-docs/src/main/sphinx/develop/verifier-mechanics.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Catalog/schema/username/password overrides are applied if specified.
+* Whitelist and blacklist are applied if specified.
+* Syntactically invalid queries are filtered.
+* Custom filters can be plugged in if needed.
+
+Query rewriting
+~~~~~~~~~~~~~~~
+* Each query is rewritten before execution for setup purposes and also to ensure that verification does not interfere with production data.
```
Comment (mayankgarg1990) [inline presto-docs/src/main/sphinx/develop/verifier-mechanics.rst]: The purpose of setup is to ensure that we do not interfere with production data. So simplify this as `Each query is rewritten before execution to ensure that verification does not interfere with production data`

File: presto-docs/src/main/sphinx/installation/verifier.rst [hunk 4]
```diff
@@ -2,9 +2,8 @@
 Presto Verifier
 ===============
 
-The Presto Verifier can be used to test Presto against another database (such as MySQL),
-or to test two Presto clusters against each other. We use it to continuously test trunk
-against the previous release while developing Presto.
+The Presto Verifier can be used to test two versions Presto against each other.
+We use it to continuously test trunk against the previous release while developing Presto.
```
Comment (rschlussel) [inline presto-docs/src/main/sphinx/installation/verifier.rst]: Who's we here? The documentation shouldn't reference facebook practice?

File: presto-docs/src/main/sphinx/develop/verifier.rst [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+Procedures and Functionalities
+------------------------------
+
+The following steps summarize the working of verifier.
+
+Importing Source Queries
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Source queries can be read through MySQL.
+* If the required schema is not available, one can also plugin a custom Provider.
```
[the remark this answers is not in the document]
Comment (caithagoras) [reply presto-docs/src/main/sphinx/develop/verifier.rst]: Re-worded

File: presto-docs/src/main/sphinx/develop/verifier.rst [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+Credentials and Filtering
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Catalog/schema/username/password overrides are applied if specified.
+* Whitelist and blacklist are applied if specified.
```
[the remark this answers is not in the document]
Comment (caithagoras) [reply presto-docs/src/main/sphinx/develop/verifier.rst]:
```
Queries within a suite can be selected and/or excluded by specifying whitelist and/or blacklist.
```

File: presto-docs/src/main/sphinx/installation/verifier.rst (hunk 4 above)
Comment (caithagoras) [reply presto-docs/src/main/sphinx/installation/verifier.rst]: Removed

File: presto-docs/src/main/sphinx/develop/verifier.rst [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+Non-Determinism
+~~~~~~~~~~~~~~~
+* In case of result mismatch due to row count mismatch or column mismatch. We rerun the control queries 2 more times.
+  Along with the initial control query results, we determine whether the control query is deterministic.
+* The 2nd additional run will only be started if the first two runs match.
+
+Automatic Failure Resolution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+* Because there are potential configuration differences between the control and the test clusters, certain queries failures are automatically resolved
+  (i.e. Global memory exceeded, Time limit exceeded).
```
[the remark this answers is not in the document]
Comment (caithagoras) [reply presto-docs/src/main/sphinx/develop/verifier.rst]: We don't have the switch yet but it make sense to me to add one.

**Comment by caithagoras** [comment]
@rschlussel What exactly are you referring to when use said "user documentation"? I though the developer guide is part of the user documentation but is more towards power users.

Currently we have a section about verifier in the `installation` section (http://prestodb.github.io/docs/current/installation/verifier.html), which contains a list of commonly-used configuration properties. How would you recommend we organize those information.

**Comment by rschlussel** [comment]
@caithagoras I didn't realize at first that it was in a section of the user documentation.  But either way my point is that all the documentation should be together in one place.  Right now there's "beginner" information in one place that isn't quite enough to understand what the verifier is or how to use it.  and then there's "advanced" information in another place that explains more background about how it works and things you can do, but doesn't tell you how to do them or where you could learn more. I think it would be much clearer for both kinds of users to have all the information in one place.  I don't think it matters whether it goes in the installation section or the developer section.

**Comment by mbasmanova** [comment]
@caithagoras This documentation would be very useful for folks who are running the verifiers and investigating the results. It is helpful to understand what kind of things are resolved automatically and what are not. Let's address the comments and land this doc.

CC: @nezihyigitbasi

**Pull request closed without merging**